From 2eebdf3f9dcad17960bf849f0d7c987de05dc612 Mon Sep 17 00:00:00 2001
Message-Id: <2eebdf3f9dcad17960bf849f0d7c987de05dc612.1346761472.git.minovotn@redhat.com>
In-Reply-To: <66350080c56dd39cd894591e70182434d5d2de75.1346761472.git.minovotn@redhat.com>
References: <66350080c56dd39cd894591e70182434d5d2de75.1346761472.git.minovotn@redhat.com>
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Mon, 3 Sep 2012 17:21:03 +0200
Subject: [PATCH 10/10] disable rdtscp on all CPU model definitions

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1346692863-562-1-git-send-email-ehabkost@redhat.com>
Patchwork-id: 41610
O-Subject: [RHEL6 qemu-kvm PATCH] disable rdtscp on all CPU model definitions
Bugzilla: 814426
RH-Acked-by: Marcelo Tosatti <mtosatti@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Gleb Natapov <gleb@redhat.com>

Bugzilla: 814426
Upstream status: not applicable
Scratch build: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=4825885

The RHEL-6 kernel KVM module does not support the rdtscp feature, and it
was never supposed to be enabled on the CPU models we have on qemu-kvm.

This should have no visible effect (and does not need machine-type
compatibility code) because all users have rdtscp disabled right now.
This simply changes the CPU model definitions to reflect reality.

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 sysconfigs/target/cpu-x86_64.conf | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 sysconfigs/target/cpu-x86_64.conf | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sysconfigs/target/cpu-x86_64.conf b/sysconfigs/target/cpu-x86_64.conf
index c6ef9d7..123a1a2 100644
--- a/sysconfigs/target/cpu-x86_64.conf
+++ b/sysconfigs/target/cpu-x86_64.conf
@@ -93,7 +93,7 @@
    stepping = "1"
    feature_edx = " sse2 sse fxsr mmx clflush pse36 pat cmov mca pge mtrr sep apic cx8 mce pae msr tsc pse de fpu"
    feature_ecx = "avx xsave aes popcnt x2apic sse4.2 sse4.1 cx16 ssse3 pclmulqdq sse3"
-   extfeature_edx = "i64 rdtscp nx syscall "
+   extfeature_edx = "i64 nx syscall "
    extfeature_ecx = "lahf_lm"
    xlevel = "0x8000000A"
    model_id = "Intel Xeon E312xx (Sandy Bridge)"
@@ -121,7 +121,7 @@
    stepping = "1"
    feature_edx = "sse2 sse fxsr mmx clflush pse36 pat cmov mca pge mtrr sep apic cx8 mce pae msr tsc pse de fpu"
    feature_ecx = "x2apic cx16 sse3"	# x2apic kvm emulated
-   extfeature_edx = "lm rdtscp fxsr mmx nx pse36 pat cmov mca pge mtrr syscall apic cx8 mce pae msr tsc pse de fpu"
+   extfeature_edx = "lm fxsr mmx nx pse36 pat cmov mca pge mtrr syscall apic cx8 mce pae msr tsc pse de fpu"
    extfeature_ecx = "svm lahf_lm"
    xlevel = "0x80000008"
    model_id = "AMD Opteron 22xx (Gen 2 Class Opteron)"
@@ -135,7 +135,7 @@
    stepping = "1"
    feature_edx = "sse2 sse fxsr mmx clflush pse36 pat cmov mca pge mtrr sep apic cx8 mce pae msr tsc pse de fpu"
    feature_ecx = "popcnt x2apic cx16 monitor sse3"	# x2apic kvm emulated
-   extfeature_edx = "lm rdtscp fxsr mmx nx pse36 pat cmov mca pge mtrr syscall apic cx8 mce pae msr tsc pse de fpu"
+   extfeature_edx = "lm fxsr mmx nx pse36 pat cmov mca pge mtrr syscall apic cx8 mce pae msr tsc pse de fpu"
    extfeature_ecx = "misalignsse sse4a abm svm lahf_lm"
    xlevel = "0x80000008"
    model_id = "AMD Opteron 23xx (Gen 3 Class Opteron)"
@@ -149,7 +149,7 @@
    stepping = "2"
    feature_edx = "sse2 sse fxsr mmx clflush pse36 pat cmov mca pge mtrr sep apic cx8 mce pae msr tsc pse de fpu"
    feature_ecx = "avx xsave aes popcnt sse4.2 sse4.1 cx16 ssse3 pclmulqdq sse3"
-   extfeature_edx = "lm rdtscp pdpe1gb fxsr mmx nx pse36 pat cmov mca pge mtrr syscall apic cx8 mce pae msr tsc pse de fpu"
+   extfeature_edx = "lm pdpe1gb fxsr mmx nx pse36 pat cmov mca pge mtrr syscall apic cx8 mce pae msr tsc pse de fpu"
    extfeature_ecx = " fma4 xop 3dnowprefetch misalignsse sse4a abm svm lahf_lm"
    xlevel = "0x8000001A"
    model_id = "AMD Opteron 62xx class CPU"
-- 
1.7.11.4

